.circleci/config.yml

```diff
@@ -61,7 +61,7 @@ commands:
 jobs:
   cannon-go-lint-and-test:
     docker:
-      - image: us-docker.pkg.dev/oplabs-tools-artifacts/images/ci-builder:latest
+      - image: us-docker.pkg.dev/${GCP_PROJECT_ID}/${GCP_ARTIFACT_REPOSITORY}/images/ci-builder:latest
```

Comment:
Project IDs are sensitive information, but are not secrets. Disclosing a Project ID does not put your account at risk. I suggest storing your project ID in the cirleci variables.

Reply:
This should be `us-docker.pkg.dev/${GCP_PROJECT_ID}/${GCP_ARTIFACT_REPOSITORY}/ci-builder:latest`. As `images` is the repository name we use.

I've setup both envs in circleci. You should be able to run CI once you've made the above changes.

Reply:
```
Starting container us-docker.pkg.dev///ci-builder:latest
  image cache not found on this host, downloading us-docker.pkg.dev///ci-builder:latest

```

can you double check the envs are there? thanks!

Reply:
Yeah, the envs are setup correctly for the project. I think the issue here is the envs can't be used in an `image` value.

Reply:
Perhaps defining them as a `parameter` would work. Then you can interpolate the values in the `image` value like:
`us-docker.pkg.dev/<<parameters.gcp_project_id>>/<<parameters.gcp_artifact_repository>>/ci-builder:latest`

Reply:
They can be read 
<img width="1262" alt="image" src="https://github.com/ethereum-optimism/optimism/assets/2582555/9a6bd1b0-8b2a-416d-bc4c-95577130eb46">

https://github.com/bobanetwork/v3-anchorage/blob/develop/.circleci/config.yml#L70

I'm just thinking what else could be a problem. Thank you for sticking with me here.

Reply:
We have those envs setup for the optimism monorepo project, rather than in a circleci context. how are your envs setup?

Reply:
I've set them here: https://app.circleci.com/settings/project/github/bobanetwork/v3-anchorage/environment-variables

Reply:
The environment variables are picked up when I trigger a circleci pipeline myself. See https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/29773/workflows/39eae03c-1b9e-4cdd-b71d-a59ae9fead8a/jobs/1406434.
Not sure if you have necessary permissions to do this, but could you try triggering a fresh new pipeline? My guess is there's some caching going on preventing the new envs from being picked up.

Reply:
Oh cool, now they're being picked up!
I could open a new PR and see what happens there? 


Additionally, is this the right repo? Or should I drop `images` now?
```
      repo:
        description: Docker repo
        type: string
        default: "${GCP_ARTIFACT_REPOSITORY}/images"
```

Also,
the **.env** file would need to be populated in order to maintain equal _local development environment_ and _GH actions_ that use docker-compose for builds (like release-docker-canary.yml).
```
- name: Build and push
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./ops/docker/Dockerfile.packages
          target: fault-mon
          push: true
          build-args:
            - ${{ env.GCP_ARTIFACT_REPOSITORY }}
            - ${{ env.GCP_PROJECT_ID }}
          tags: ${{ env.ORG }}/fault-mon:${{ needs.canary-publish.outputs.canary-docker-tag }}
```

docker-compose.yml
```
  l2:
    build:
      context: .
      dockerfile: Dockerfile.l2
      args:
         - GCP_PROJECT_ID=${GCP_PROJECT_ID:-populate-manually}
         - GCP_ARTIFACT_REPOSITORY=${GCP_ARTIFACT_REPOSITORY:-oplabs-tools-artifacts}
```

Reply:
You can drop `images` so the repo is `${GCP_ARTIFACT_REPOSITORY}`.

> Also,
the .env file would need to be populated in order to maintain equal local development environment and GH actions that use docker-compose for builds (like release-docker-canary.yml).

Good callout. Not exactly sure how this is setup. I'll get back to you on whether we can update the gh envs.

Reply:
Cool, will do. And let me know on other stuff. Thanks! 🙏🏻